src/frontend/app/components/ScenarioContent.tsx

```diff
@@ -36,7 +40,9 @@ export default function ScenarioContent({ scenario }: { scenario: Scenario }) {
         <ScenarioContentBody scenario={scenario} />
       </div>
       <Link
-        href={page === 'home' ? '/' : '/saved'}
+        href={
+          page === 'home' ? `/?${searchParams.toString()}` : `/saved?${searchParams.toString()}`
```

Comment:
We don't have search or filters on saved page is it neccessary to keep them there?